internal/internal_workflow.go

```diff
@@ -1436,6 +1546,29 @@ func (h *queryHandler) execute(input []interface{}) (result interface{}, err err
 	return executeFunction(h.fn, input)
 }
 
+func (h *updateHandler) validate(input []interface{}) (err error) {
+	defer func() {
+		if p := recover(); p != nil {
```

Comment:
I wonder if we should have some worker-level option of whether to treat validator panics as persisted or not. We have `worker.Options.WorkflowPanicPolicy` that defines how to treat workflow panics, so an `UpdateValidatorPanicPolicy` may make sense. I could argue for either way on what the default should be though.

Actually, now that I think about it, I guess we can keep this (default durable) and if they don't like that, they can recover themselves and make it non-durable.

Reply:
There's also the errors that come from the inability to deser commonpb.Payloads or to invoke the registered validator/executor funcs .... the user doesn't have any control over those and they default to bypass. I'm not sure whether we need an option to control that bypass/nobypass or not.

Reply:
In retrospect, I agree. If they want panics to not be persisted, they can recover from them themselves. Panics shouldn't happen anyways.

Reply:
In a sense you want to bypass for those so that someone/something can't flood you with garbage input as a DoS. But then again it might be useful in a development setting to capture that info for debugging. This duality is what let me toward a workflow option. If we don't want an option then forcing bypass is probably safer.

Reply:
> This duality is what let me toward a workflow option. If we don't want an option then forcing bypass is probably safer.

Agreed. You could consider a deserialization failure as a panic from the "unexpected input" POV. Otherwise, if you craft a panic-causing input, the DoS is still a thing.

Reply:
Current direction is that any error or panic from validation is implicitly prefer-bypass so at this point a preference would not be useful.